assets/actions/package.go

```diff
@@ -0,0 +1,257 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"log"
+	"io/ioutil"
+	"strings"
+	"net/http"
+	"time"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/layer5io/meshkit/utils/catalog"
+	"github.com/layer5io/meshkit/models/catalog/v1alpha1"
+)
+
+// CatalogPattern represents the catalog pattern structure
+type CatalogPattern struct {
+	ID           string `json:"id"`
+	Name         string `json:"name"`
+	PatternFile  string `json:"pattern_file"`
+	CatalogData struct {
+        PatternInfo   string        `json:"pattern_info"`
```

Comment:
Please use the `catalog data` struct defined in meshkit (v1alpha2).

Same use meshery pattern from `meshery server` instead of redefining.